sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -92,7 +93,38 @@ class QueryExecution(
       sparkSession.sessionState.analyzer.executeAndCheck(logical, tracker)
     }
     tracker.setAnalyzed(plan)
-    plan
+
+    mode match {
```

Comment:
This part will have to evolve. There are 3 options for us to consider.

1. Add a special type of commands that must be executed during the analysis (i.e. generalize and polish this PR).
2. Add a special mix-in interface for procedures that know the type of the last result set before the execution. All other procedures will not output anything if invoked via `spark.sql`.
3. Migrate to `qe.commandExecuted` instead of `qe.analyzed` everywhere as we will know the output only after executing the procedure.

Reply:
@cloud-fan, any thoughts?